src/components/CompareResults/SearchInput.tsx

```diff
@@ -1,53 +1,86 @@
+import { useState, useCallback } from 'react';
+
 import Close from '@mui/icons-material/Close';
 import SearchIcon from '@mui/icons-material/Search';
-import { FormControl, InputAdornment, TextField, Button } from '@mui/material';
-import { style } from 'typestyle';
-
-import { Spacing } from '../../styles';
-
-const styles = {
-  container: style({
-    // TODO: to be added at the implementation time
-    display: 'none',
-    marginRight: 'auto',
-    $nest: {
-      '.form-container': {
-        maxWidth: '447px',
-      },
-      '.MuiButtonBase-root': {
-        marginLeft: Spacing.Small,
-      },
-    },
-  }),
-};
-
-function SearchInput() {
+import {
+  InputAdornment,
+  TextField,
+  Button,
+  Box,
+  IconButton,
+} from '@mui/material';
+
+import { Strings } from '../../resources/Strings';
+import { simpleDebounce } from '../../utils/simple-debounce';
+
+interface SearchInputProps {
+  onChange: (searchTerm: string) => unknown;
+}
+function SearchInput({ onChange }: SearchInputProps) {
+  const [searchTerm, setSearchTerm] = useState('');
+
+  // By using useCallback, we ensure that we always have the same instance of
+  // onDebouncedChange, so that when calling `clear` the timeout will always be
+  // appropriately cleared.
+  const onDebouncedChange = useCallback(simpleDebounce(onChange), [onChange]);
+
+  const onValueChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.currentTarget.value;
+    setSearchTerm(value);
+    onDebouncedChange(value);
+  };
+
+  const clearSearchTerm = () => {
+    setSearchTerm('');
+    onDebouncedChange.clear();
+    onChange('');
+  };
+
+  const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    onDebouncedChange.clear();
+    onChange(searchTerm);
+  };
+
+  const inputStrings = Strings.components.searchResultsInput;
+
   return (
-    <div className={styles.container} data-testid={'search-by-title-test-name'}>
-      <div className='form-container'>
-        <FormControl variant='outlined'>
-          <TextField
-            disabled
-            placeholder='Search by title'
-            id='search-revision-input'
-            size='small'
-            InputProps={{
-              startAdornment: (
-                <InputAdornment position='start'>
-                  <SearchIcon />
-                </InputAdornment>
-              ),
-              endAdornment: (
-                <InputAdornment position='end'>
-                  <Close />
-                </InputAdornment>
-              ),
-            }}
-          />
-        </FormControl>
-      </div>
-      <Button disabled>Apply</Button>
-    </div>
+    <Box
+      component='form'
+      aria-label={inputStrings.label}
+      onSubmit={onSubmit}
+      sx={{ display: 'flex', marginRight: 'auto' }}
+    >
+      <TextField
+        placeholder={inputStrings.placeholder}
```

Comment:
What do you say about using a title or something similar so that the user can see the options he can search by?
The placeholder is not fully visible. I could only see `title, platform and revision` when testing. I wasn't sure if I can search by `options`.

Reply:
Yeah I was wondering about adding a title too... It might be good to ask the a11y team about that, as a follow-up. Currently it's in an aria-label but it could easily be a title instead...

I think we should make everything the user might want to search for work so that they don't have to wonder about that. I'll replace the placeholder with "Search results".